### PR TITLE
add Parser [all fixes available in update-parser branch]

### PR DESCRIPTION
**Pull request by Ryu-ko**
I added a folder with Parser and fixed an error with buttons in MainWindow.xaml.cs  in numberButton_Click()

### CODE REVIEW

File: calculator/calculator/calculator/ParserF/Expression.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            Value = value;
+        }
+
+        public override string ToString()
+        {
+            return Value.ToString();
+        }
+    }
+
+    public class BinaryExpression : Expression //само выражение
```
Comment (mebry) [inline calculator/calculator/calculator/ParserF/Expression.cs:25]: Write in English

File: calculator/calculator/calculator/Views/MainWindow.xaml.cs [hunk 2]
```diff
@@ -81,12 +84,33 @@ public static ObservableCollection<Function> GetFunctions()
...
             variablesView.ItemsSource = DataVault.getVariables();
+        }
+
+        private void Calculate_Click( object sender, RoutedEventArgs e )
+        {
+            try
+            {
+                var input = inputField.Text;
+                var expression = parser.Parse(input);
+                //Console.WriteLine($"Parsed Expression Tree: {expression}");
```
Comment (mebry) [inline calculator/calculator/calculator/Views/MainWindow.xaml.cs:104]: If you don't need this code, delete it

File: calculator/calculator/calculator/ParserF/Computations.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                    case "+":
+                        return left + right;
+                    case "-":
+                        return left - right;
+                    case "*":
+                        return left * right;
+                    case "/":
+                        return left / right;
+                    default:
+                        throw new Exception("Unknown operator: " + binaryExpr.Operator);
```
Comment (mebry) [inline calculator/calculator/calculator/ParserF/Computations.cs:33]: it is better to use more specific types of exceptions

File: calculator/calculator/calculator/ParserF/Parser.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        {
+            var left = ParseOperand();
+
+            while (position < tokens.Count && (tokens[position].Value == "*" || tokens[position].Value == "/"))
+            {
+                var op = tokens[position++].Value;
+                var right = ParseOperand();
+                left = new BinaryExpression(left, right, op);
+
+                Console.WriteLine($"Parsed Term: {left}");
```
Comment (mebry) [inline calculator/calculator/calculator/ParserF/Parser.cs:57]: clean the code from such code, especially in cycles

File: calculator/calculator/calculator/ParserF/Token.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+{
+    public class Token
+    {
+        public enum TokenType
+        { 
+            Number,
+            Operator,
+            LeftParen,
+            RightParen
+        }
```
Comment (mebry) [inline calculator/calculator/calculator/ParserF/Token.cs:17]: You can put it in a separate file

**Review by mebry — CHANGES_REQUESTED** [review]

File: calculator/calculator/calculator/ParserF/Expression.cs (hunk 1 above)
Comment (Ryu-ko) [reply calculator/calculator/calculator/ParserF/Expression.cs:25]: Fixed in branch UpdateParser

File: calculator/calculator/calculator/Views/MainWindow.xaml.cs (hunk 2 above)
Comment (Ryu-ko) [reply calculator/calculator/calculator/Views/MainWindow.xaml.cs:104]: Fixed in branch UpdateParser

File: calculator/calculator/calculator/ParserF/Computations.cs (hunk 3 above)
Comment (Ryu-ko) [reply calculator/calculator/calculator/ParserF/Computations.cs:33]: I added new class for Exceptions and change the same lines. We will improve Exception handling later. (changes in branch UpdateParser)

File: calculator/calculator/calculator/ParserF/Parser.cs (hunk 4 above)
Comment (Ryu-ko) [reply calculator/calculator/calculator/ParserF/Parser.cs:57]: Fixed in branch UpdateParser

File: calculator/calculator/calculator/ParserF/Token.cs (hunk 5 above)
Comment (Ryu-ko) [reply calculator/calculator/calculator/ParserF/Token.cs:17]: Fixed in branch UpdateParser

**Pull request merged**